text/0003-blockspace-regions.md

```diff
@@ -0,0 +1,252 @@
+    // The assignee of the region is the parachain the region gives the right to create blocks to.
+    assignee: ParaId,
+}
+```
+
+Regions each have a 256-bit identifier, which are unique within the branch of the  relay-chain they are created.
+
+```rust
+type RegionId = [u8; 32];
+```
+
+The `RATE_DENOMINATOR = 57600` is used to provide a fixed reference point for resource consumption rates and levels. It is sufficiently large as to allow blocks to come extremely infrequently (~once per week with a numerator of 1) but not so large as to be unwieldy in 64-bit integers. 57600 is a deliberately chosen, very composite number. This makes it possible to find exact fractions for common desired ratios of the relay-chain block-time such as 1/3, 1/4, 1/5, and so on. It also divides cleanly into `28 * DAYS`.
+
+### The Regions Pallet
+
+Regions will be managed within a "Regions" Pallet which exposes the following storage and `Call`s:
+
+```rust
+// Updated whenever regions are created, modified, or collected.
+storage map Regions: RegionId -> Region;
+
+// Updated when regions are created, transfered, or collected.
+// This is required for runtime APIs or other higher-level logic on collators to iterate the regions assigned
+// to a specific para.
+storage double_map RegionsPerPara: (ParaId, RegionId) -> ();
+
+// All functions are gated to permissioned origins, which are controlled by governance and intended to be assigned to system chains for managing regions.
+
+// Create a region. This is gated on allowed origins, e.g. a `RegionCreator` origin, set by governance.
+//
+// This inserts the region into the `RegionsPerPara` and `Regions` maps.
+//
+// If the region's `end` is `Some`, this sets up garbage collection logic for the region.
+fn create(Origin, Region) -> RegionId;
+
+// Update the end-point of a region. This is gated on allowed origins, e.g. a `RegionCreator` origin, set by governance.
+//
+// This is a no-op if the region's end is already `Some` or there is no region with that ID.
+// Otherwise, this sets up garbage-collection logic for the region. If `end` is in the past, `end` is set to the current block number.
+fn set_end(Origin, RegionId, end: u32);
+```
+
+This API is intended to be wrapped by logic implementing interfaces such as RFC-5.
+
+Created regions for a core MUST NOT exceed a combined block rate of 1 at any block.
+
+### Changes to backing/availability
+
+Regions are used to modify the behavior of the parachain backing/availability pipeline. The first major change is that parachain candidates submitted to the relay-chain in the `ParasInherent` will be annotated with the `RegionId` that they are intended to occupy. Validators and collators do the work of figuring out which blocks are assigned to which regions, lifting the burden of granular scheduling off of the relay chain.
+
+The **maximum consumption** of a region at any block number `now` is given by:
+`maximum_consumption(now, region) = min(now, end) - start * rate`
```

Comment:
Shouldn't there be a pair of braces or I'm not getting this?
`maximum_consumption(now, region) = (min(now, end) - start) * rate`